cmd/root.go

```diff
@@ -117,9 +117,40 @@ func MakeCommands() *cobra.Command {
 	// just the error message.
 	rootCmd.SilenceUsage = true
 
+	setFlagErrorFuncAndValidateArgs(rootCmd)
+
 	return rootCmd
 }
 
+func setFlagErrorFunc(cmd *cobra.Command, err error) error {
+	if e := cmd.Help(); e != nil {
+		return e
+	}
+	return err
+}
+
+func setFlagErrorFuncAndValidateArgs(command *cobra.Command) {
+	visitAll(command, func(cmd *cobra.Command) {
+		cmd.SetFlagErrorFunc(setFlagErrorFunc)
+
+		if cmd.Args == nil {
+			return
+		}
+
+		cmdArgs := cmd.Args
+		cmd.Args = func(cccmd *cobra.Command, args []string) error {
+			if err := cmdArgs(cccmd, args); err != nil {
+				if e := cccmd.Help(); e != nil {
+					return e
```

Comment:
So if I understand this correctly, it's rewriting each command with a post-error hook, and returning the help instead of the error, if there is one. Wouldn't this discard potentially useful error messaging in `err`?

Reply:
If for some odd reason printing the `Help()` message returns an error we want to return that, but we could probably wrap those together.